backend/src/services/interrupt.py

```
@@ -0,0 +1,438 @@
+    - Nonce validation for replay protection
+    - Thread ownership verification
+    - Decision processing via strategy handlers
+    """
+
+    # Handler registry
+    _handlers: Dict[DecisionType, type] = {
+        DecisionType.APPROVE: ApproveHandler,
+        DecisionType.EDIT: EditHandler,
+        DecisionType.REJECT: RejectHandler,
+        DecisionType.RESPOND: RespondHandler,
+    }
```

Ruff (0.14.11)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)